Updater component

rmkraus opened the pull request:
Added component to indicate when an update is available on pypi.

rmkraus committed:
Created updater component

rmkraus committed:
Added friendly name to updater component.

rmkraus committed:
Updating fronted to latest version.

rmkraus committed:
Merge remote-tracking branch 'balloob/dev' into update_notify

Conflicts:
	homeassistant/components/frontend/version.py
	homeassistant/components/frontend/www_static/frontend.html

balloob commented on homeassistant/components/updater.py:
```diff
@@ -0,0 +1,51 @@
+    return True
+
+
+def get_newest_version():
```
Can you wrap the whole method in a `try` block?

- For exceptions from requests, please catch `requests.RequestException`
- For JSON parse errors from `json()`, catch `ValueError`
- For errors related to JSON object not being expected shape, catch `KeyError`.

balloob commented on homeassistant/components/updater.py:
```diff
@@ -0,0 +1,51 @@
+    try:
+        req = requests.get(PYPI_URL)
+    except OSError:
+        _LOGGER.warning('Could not contact PyPI to check for updates')
```
Could you change this to `_LOGGER.exception` so we know what is going wrong?

balloob commented on homeassistant/components/updater.py:
```diff
@@ -0,0 +1,51 @@
+                {'friendly_name': 'Update Available'})
+
+    event.track_time_change(hass, check_newest_version,
+                            hour=12, minute=0, second=0)
```
You can pass in lists for time keywords.
```python
hour=[0, 12], minute=0, second=0)
```

balloob commented on homeassistant/components/updater.py:
```diff
@@ -0,0 +1,51 @@
+        newest = get_newest_version()
+        if newest != CURRENT_VERSION and newest is not None:
+            hass.states.set(
+                '{}.Update'.format(DOMAIN), newest,
```
Entity ID is static, can you move it to a constant. Also, entity ids should be lowercase.

balloob commented on homeassistant/components/updater.py:
```diff
@@ -0,0 +1,51 @@
+        if newest != CURRENT_VERSION and newest is not None:
+            hass.states.set(
+                '{}.Update'.format(DOMAIN), newest,
+                {'friendly_name': 'Update Available'})
```
Can you use `ATTR_FRIENDLY_NAME` from `homeassistant.const`

rmkraus committed:
Implemented suggestions from Paulus for updater

1) Better Error handling when making PyPI requests.
2) More efficient event scheduling.
3) ENTITY_ID in constant
3) friendly_name from constant

rmkraus committed:
Updater suggestions from Paulus

1) Moved error checking into get_newest_version function.
2) Fixed import formatting mistake.

balloob commented:
Looks good :dolphin: 

Merge when tests pass.

rmkraus committed:
Using ATTR_FRIENDLY_NAME in updater component.

rmkraus committed:
Fixed typo in updater entity attributes.

Left some quotes in there. My bad.
